libs/server/StoreWrapper.cs

```diff
@@ -693,15 +693,19 @@ private async Task InitiateCheckpoint(bool full, CheckpointType checkpointType,
                     objectStoreCheckpointResult = await objectStore.TakeHybridLogCheckpointAsync(checkpointType, tryIncremental);
             }
 
-            // If cluster is enabled the replication manager is responsible for truncating AOF
-            if (serverOptions.EnableCluster && serverOptions.EnableAOF)
+            // Only truncate AOF if both stores are checkpointed
+            if (storeType == StoreType.All)
```

Comment:
The logic should be slightly more complex, since some people might be running Garnet with only the main store which means we can truncate the log safely even if the storeType == StoreType.MainStore